src/presign/participant.rs

```diff
@@ -399,7 +558,22 @@ impl PresignParticipant {
             ));
         }
 
-        Ok(ret_messages)
+        // Additionally, handle any round 3 messages which may have been received too
+        // early
+        let mut presign_record = None;
+        let retrieved_messages = self.fetch_messages(
+            MessageType::Presign(PresignMessageType::RoundThree),
+            message.id(),
+        )?;
+        for msg in retrieved_messages.iter() {
+            let (pr, _) = self.handle_round_three_msg(rng, msg, main_storage)?;
+            // pr will be Some at most once
```

Comment:
Could we add an explicit runtime check to ensure this comment is true? While this is true now, with refactor and other code changes, assumptions like this can be invalidated. It would be good to have a check if it ever happens.